recipes/instinct-cpp/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("hash-library/8.0", transitive_headers=True)
+        self.requires("bshoshany-thread-pool/4.1.0", transitive_headers=True)
+
+        if self.options.with_duckdb:
+            self.requires("duckdb/0.10.2", transitive_headers=True)
+
+        self.requires("uriparser/0.9.7", transitive_headers=True)
+        self.requires("crossguid/0.2.2", transitive_headers=True)
+        self.requires("protobuf/5.27.0", transitive_headers=True)
+        self.requires("reactiveplusplus/2.1.1", transitive_headers=True)
+        self.requires("icu/74.1")
+        self.requires("tsl-ordered-map/1.1.0", transitive_headers=True)
+        self.requires("fmt/10.2.1", transitive_headers=True)
+        self.requires("fmtlog/2.2.1", transitive_headers=True)
+        self.requires("nlohmann_json/3.11.3", transitive_headers=True)
+        self.requires("base64/0.5.2", transitive_headers=True)
+        self.requires("libcurl/8.6.0", transitive_headers=True)
+        self.requires("inja/3.4.0", transitive_headers=True)
+        self.requires("concurrentqueue/1.0.4", transitive_headers=True)
+        self.requires("cpptrace/0.5.4", transitive_headers=True)
+
+        if self.options.with_pdfium:
+            self.requires("pdfium/95.0.4629", transitive_headers=True)
+
+        if self.options.with_duckx:
+            self.requires("duckx/1.2.2", transitive_headers=True)
+
+        if self.options.with_exprtk:
+            self.requires("exprtk/0.0.2", transitive_headers=True)
+
+        self.requires("llama-cpp/b3040", transitive_headers=True)
+        self.requires("cpp-httplib/0.15.3", transitive_headers=True)
+        self.requires("cli11/2.4.1")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+        
+        def _ensure_dep_has_option(dep, option, value):
+            if self.dependencies[dep].options[option] != value:
+                raise ConanInvalidConfiguration(f'{self.ref} needs -o="{dep}/*:{option}={value}"')
+
+        _ensure_dep_has_option("duckdb", "with_httpfs", self.default_options["duckdb/*:with_httpfs"])       
+        _ensure_dep_has_option("icu", "with_extras", self.default_options["icu/*:with_extras"])
+        _ensure_dep_has_option("icu", "data_packaging", self.default_options["icu/*:data_packaging"])
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
```

Comment:
A bit suprising for a library with so many dependencies to have these little amount of options, it's great! :)